packages/react-router/lib/components.tsx

```diff
@@ -631,6 +635,7 @@ export function createRoutesFromChildren(
       loader: element.props.loader,
       action: element.props.action,
       errorElement: element.props.errorElement,
+      hasErrorBoundary: element.props.errorElement != null,
```

Comment:
`createRoutesFromChildren` fills this field in for you automatically based on `errorElement`.  This means the user-facing JSX API doesn't change at all.  

If a user decides to create their own `routes` array manually and do `<DataBrowserRouter routes={routes} />`, then they are responsible for including `hasErrorBoundary` accordingly.  This feels sort of annoying but seems necessary unless we want to walk the tree and add it on our own prior to handing off to `createRouter` - which means we need to create copies of the routes to avoiding mutating in place, etc.

@ryanflorence WDYT?

Reply:
Yeah I'd like to add it before handing it off to `createRouter`. That shouldn't be a detail React Router users should have to care about.

Reply:
ok cool - added that in https://github.com/remix-run/react-router/pull/9141/commits/11b9bb6fab9cad48601accc7c9cdc3d3ad3054a5

Reply:
and renamed to `enhanceManualRouteObjects` in [b45f4f5](https://github.com/remix-run/react-router/pull/9141/commits/b45f4f597d46a0671b91886813b26ef0fa58b664) 😄